### PR TITLE
Additional Turret Options

### PR DESCRIPTION
**Pull request by Hazeofdream**
had this on my drive for awhile and figure I'd throw this out if any dev/contributor is interested in this.

tl;dr adds blast radius and blast damage to turrets so people don't have to cheese wire explosives/prop explosives for their bullet based contraptions

Adds Convars for these values since 1) hardlocking/clamping like the damage was done is gross and 2) it's a pretty uh, *colorful function*

Not immediately suggesting to merge this as I haven't figured out how to maintain old turrets as this would modify it's inputs, and even duplicating it would mean having to redo turrets just based on whether servers had this. I GUESS that would be the best option if their is no alternative

obligatory not a good coder probably has some unncessary/dogshit code in it but lmao i tried

### CODE REVIEW

File: lua/entities/gmod_wire_turret.lua [hunk 1]
```diff
@@ -1,182 +1,219 @@
...
+	self:SetDamage(damage)
+	self:SetSpread(spread)
+	self:SetBlastDamage(blastdamage)
+	self:SetBlastRadius(blastradius)
+	self:SetTracer(tracer)
+	self:SetTraceNum(tracernum)
+	self:SetNumBullets(numbullets)
+end
+
+-- Legacy Support
```
Comment (Vurv78) [inline lua/entities/gmod_wire_turret.lua:205]: Lua doesn't have any sort of overloads. This just replaces the previous function `Setup` with a different one that takes the old parameters

File: lua/entities/gmod_wire_turret.lua [hunk 2]
```diff
@@ -1,182 +1,219 @@
...
+	self.spreadvector.x = self.spread
+	self.spreadvector.y = self.spread
+end
+
+function ENT:SetDamage( damage )
+	self.damage = math.Clamp( damage, 0, 100 )
+end
+
+function ENT:SetBlastDamage( blastdamage )
+	self.blastdamage = math.Clamp( blastdamage, 0, GetConVar("wire_turret_blastdamage_max") )
```
Comment (Vurv78) [inline lua/entities/gmod_wire_turret.lua:153]: `GetConVar` returns the actual `ConVar` object. You want either `GetConVar(...):GetInt()` or `GetConVarInt(...)`.

Ideally you'd keep the convar outside of the function, and then just call GetInt on that whenever SetBlastDamage is called

File: lua/entities/gmod_wire_turret.lua [hunk 3]
```diff
@@ -1,182 +1,219 @@
...
+end
+
+function ENT:SetBlastRadius( blastradius )
+	self.blastradius = math.Clamp( blastradius, 0, GetConVar("wire_turret_blastradius_max") )
```
Comment (Vurv78) [inline lua/entities/gmod_wire_turret.lua:157]: Same as above

File: lua/entities/gmod_wire_turret.lua [hunk 4]
```diff
@@ -1,182 +1,219 @@
...
+	self:SetSound(sound)
+	self:SetDamage(damage)
+	self:SetSpread(spread)
+	self:SetTracer(tracer)
+	self:SetTraceNum(tracernum)
+	self:SetNumBullets(numbullets)
+end
+
+duplicator.RegisterEntityClass( "gmod_wire_turret", WireLib.MakeWireEnt, "Data", "delay", "damage", "force", "sound", "numbullets", "spread", "blastdamage", "blastradius", "tracer", "tracernum" )
+-- Legacy Support
```
Comment (Vurv78) [inline lua/entities/gmod_wire_turret.lua:218]: Once again not how this works, pretty sure this would just overwrite the other one.

**Review by Vurv78 — CHANGES_REQUESTED** [review]
Haven't tested in game but these are my reviews.

**Comment by Vurv78** [comment]
I'm not really sold on the idea right now. I think maybe this could be its own component, like `gmod_wire_cannon` or something to really make it obvious that it blows stuff up.

Also so that server owners can properly restrict it, since I think people would want to restrict a fully automatic exploding bullet weapon a lot more than just a typical bullet turret.

File: lua/entities/gmod_wire_turret.lua (hunk 4 above)
Comment (Vurv78) [reply lua/entities/gmod_wire_turret.lua:218]: If you want to fix this stuff and have proper legacy support you need to put the blast damage / radius inputs after the trace ones

**Comment by thegrb93** [comment]
Seems mingey

**Comment by Hazeofdream** [comment]
> Seems mingey
_laughs in wire explosive_

**Comment by thegrb93** [comment]
Well, a lot of servers restrict explosives. Now you've added something else for them to have to restrict

**Comment by thegrb93** [comment]
I like that wire_cannon idea. Especially if it is like a modifiable acf cannon.

**Comment by Hazeofdream** [comment]
sounds like a seperate wire turret with extra steps

**Comment by Hazeofdream** [comment]
I'll try the cannon idea but leave this here for now, It probably wont be the most advanced thing either tho so don't expect acf levels of customization.
